playground/infrastructure/cd_helper.py

```diff
@@ -0,0 +1,165 @@
+        code_path = self._get_cloud_file_name(sdk=example.sdk, base_folder_name=object_meta["name"],
+                                              file_name=object_meta["name"])
+        output_path = self._get_cloud_file_name(sdk=example.sdk, base_folder_name=object_meta["name"],
+                                                file_name=object_meta["name"], extension="output")
+        meta_path = self._get_cloud_file_name(sdk=example.sdk, base_folder_name=object_meta["name"],
+                                              file_name="meta", extension="info")
+        file_names[code_path] = example.code
+        file_names[output_path] = example.output
+        file_names[meta_path] = str(object_meta)
+        for file_name, file_content in file_names.items():
+            local_file_path = os.path.join(TEMP_FOLDER, example.pipeline_id, file_name)
+            with open(local_file_path, 'w') as file:
+                file.write(file_content)
+            file_names[file_name] = local_file_path  # don't need content anymore, instead save the local path
+        return file_names
+
+    def _get_data_from_template(self, code):
+        """
+        Find beam-playground tag and extract the information of the example to dictionary
+        Args:
+            code: code of an example
+
+        Returns: dictionary with name, description, categories, etc of the source code
+
+        """
+        res = re.search(PATTERN, code)
+        if res is not None and res.span() is not None:
+            m = res.span()
+            yaml_code = code[m[0]:m[1]]
+            try:
+                object_meta = yaml.load(yaml_code, Loader=yaml.SafeLoader)
+                return object_meta[PLAYGROUND_TAG]
+            except Exception as exp:
+                print(exp)  ## todo add logErr
+
+    def _get_cloud_file_name(self, sdk: Sdk, base_folder_name: str, file_name: str, extension: str = None):
+        """
+        Get the path where file will be stored at the bucket.
+        Args:
+            sdk: sdk of the example
+            file_name: name of the example
+            base_folder_name: name of the folder where example is stored (eq. to example name)
+            extension: extension of the file
+
+        Returns: file name
+        """
+        if extension is None:
+            extension = EXTENSIONS[Sdk.Name(sdk)]
+        return os.path.join(Sdk.Name(sdk), base_folder_name, "{}.{}".format(file_name, extension))
+
+    def _upload_blob(self, source_file: str, destination_blob_name: str):
+        """
+        Upload a file to the bucket.
+        Args:
+            source_file: name of the file to be stored
+            destination_blob_name: "storage-object-name"
+        """
+        os.environ[
+            'GOOGLE_APPLICATION_CREDENTIALS'] = "/Users/dariamalkova/IdeaProjects/beam/playground/infrastructure/play-test-key.json"
+        storage_client = storage.Client()
+        bucket = storage_client.bucket(BUCKET_NAME)
```

Comment:
Let's initialize the storage client and bucket once

Reply:
right